osu.Game/Online/Leaderboards/LeaderboardScore.cs

```diff
@@ -377,6 +388,37 @@ public RankLabel(int? rank)
             public LocalisableString TooltipText { get; }
         }
 
+        private class DateLabel : DrawableDate
+        {
+            public DateLabel(DateTimeOffset date)
+                : base(date)
+            {
+                Font = OsuFont.GetFont(size: 17, weight: FontWeight.Bold, italics: true);
+            }
+
+            protected override string Format()
+            {
+                var now = DateTime.Now;
+                var difference = now - Date;
+
+                // TODO(optional): support localisation (probably via `CommonStrings.CountHours()` etc.)
+                // requires pluralisable string support framework-side
+
+                if (difference.TotalSeconds < 10)
```

Comment:
I agree that this should be sharing the exact implementation that is used in https://github.com/ppy/osu/blob/34135138b66a19ce00e676df7c83361ad4154304/osu.Game/Overlays/BeatmapSet/Scores/ScoreboardTime.cs (and show show at all times, rather than have a cutoff).

Reply:
I will try to merge these two implementations, but I still have to separate the formats.
I'll probably keep using the current timespan format, since there's very little room in the current design.

Reply:
So do you prefer that the formats used in both are identical? Or should I create a struct that specifies what strings to use for seconds, minutes, days, etc.?

Reply:
The format used in the link above is only at most a character or two longer than yours. How is "little room" in current design an issue? Do you have screenshots of actual problematic cases?

Reply:
By little room I meant the `X days ago`, etc. format used in the design document.

Reply:
What "design document" is being referred to here? Do you mean either of these links?

https://github.com/ppy/osu/blob/34135138b66a19ce00e676df7c83361ad4154304/osu.Game/Overlays/BeatmapSet/Scores/ScoreboardTime.cs#L26-L27

If it is one of these links, then the intent was to *cross-reference* the two links against each other, rather than using something verbatim from the momentjs docs. Please open the rankings overlay or the [relevant test scene](https://github.com/ppy/osu/blob/34135138b66a19ce00e676df7c83361ad4154304/osu.Game.Tests/Visual/Online/TestSceneScoreboardTime.cs) and see how the component being linked to actually works and what is the format that it is printing dates in.

Reply:
I guess not really a document, but the screenshot here https://github.com/ppy/osu/issues/8435#issuecomment-705718759